internal/transport/handler_server_test.go

```diff
@@ -277,14 +303,16 @@ func (s) TestHandlerTransport_HandleStreams(t *testing.T) {
 		func(ctx context.Context, method string) context.Context { return ctx },
 	)
 	wantHeader := http.Header{
-		"Date":         nil,
-		"Content-Type": {"application/grpc"},
-		"Trailer":      {"Grpc-Status", "Grpc-Message", "Grpc-Status-Details-Bin"},
-		"Grpc-Status":  {"0"},
+		"Date":          {},
+		"Content-Type":  {"application/grpc"},
+		"Trailer":       {"Grpc-Status", "Grpc-Message", "Grpc-Status-Details-Bin"},
+		"Custom-Header": {"Custom header value", "Another custom header value"},
 	}
-	if !reflect.DeepEqual(st.rw.HeaderMap, wantHeader) {
```

Comment:
Note that `st.rw.HeaderMap` is not only deprecated but also might contain values that are different from `st.rw.Result().Header` - for example, it would contain the header values from the second `SendHeader` invocation, even though the `ht.do` func arguments are executed strictly sequentially and in-order.